app/controllers/miq_action_controller.rb

```diff
@@ -332,6 +287,7 @@ def action_build_edit_screen
   # Build the alert choice hash for evaluate_alerts action_type
   def action_build_alert_choices
     @edit[:choices] = MiqAlert.all.each_with_object({}) { |a, h| h[a.description] = a.guid } # Build the hash of alert choices
+    @edit[:allChoices] = MiqAlert.all.each_with_object({}) { |a, h| h[a.description] = a.guid }
```

Comment:
isn't this the same as the line above?  why do we need the same value in `@edit[:choices]` and in `@edit[:allChoices]`?

Reply:
When editing the form, `@edit[:choices]` only pass the unselected alerts to [edit.html.haml](https://github.com/ManageIQ/manageiq-ui-classic/blob/cf5e5fb6ad7dacd8a628d9a372a48cc6ec22de4f/app/views/miq_action/edit.html.haml) because selected alerts are already deleted [here](https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/miq_action_controller.rb#L302). It would make dual list select right side empty because selected alerts information are not passed in. 
![Screen Shot 2021-08-02 at 12 36 30 PM](https://user-images.githubusercontent.com/82469658/127894938-8a973e45-5f0f-43a0-9a5d-110a977e45d6.png)

I chose keep an unmodified version of the entire alert list using `@edit[:allChoices] `so that it could pass both selected and unselected alerts into edit form. And I can refer back selected alerts [here](https://github.com/ManageIQ/manageiq-ui-classic/pull/7793) to make sure the selected value are showing up in the right hand side of the dual list. Not sure if this is the best practice though. 
![Screen Shot 2021-08-02 at 12 48 15 PM](https://user-images.githubusercontent.com/82469658/127896255-6c5aac44-8797-4ad1-abad-bda592730562.png)

Reply:
maybe then it should be initialized like this:
```
@edit[:allChoices] = @edit[:choices].dup
```

However, I still think these hash key names are not obvious (and are in different styles).

Reply:
I updated to `@edit[:allAlertChoices]` I am wondering whether you could take a look at it. Thank you!

Reply:
this is better.  Thanks for updating.